### PR TITLE
Fix gpcheckcat 'oid does not exist' error for catalog tables

### PR DESCRIPTION
**Pull request by ShirishaRao**
**Issue:**
gpcheckcat reports column "oid" does not exist error while running ‘missing_extraneous’ test on pg_shdescription catalogue table.

```
Connected as user 'sshirisha' to database 'miss_attr_db5', port '7000', gpdb version '7.0'
      -------------------------------------------------------------------
      Batch size: 4
      Performing test 'unique_index_violation'
      Total runtime for test 'unique_index_violation': 0:00:01.62
      Performing test 'duplicate'
      Total runtime for test 'duplicate': 0:00:16.29
      Performing test 'missing_extraneous'
        Execution error: column "oid" does not exist
      LINE 4:                 SELECT oid FROM pg_shdescription
                                     ^


                SELECT oid
                FROM (
                      SELECT oid FROM pg_shdescription
                      WHERE objoid='18362' and classoid='1213'
                      UNION ALL
                      SELECT oid FROM gp_dist_random('pg_shdescription')
                      WHERE objoid='18362' and classoid='1213'
                ) alloids
                GROUP BY oid
                ORDER BY count(*) desc, oid

      Total runtime for test 'missing_extraneous': 0:00:00.54
      Performing test 'inconsistent'
      Total runtime for test 'inconsistent': 0:00:15.63
      Performing test 'foreign_key'
      Total runtime for test 'foreign_key': 0:00:02.67
      Performing test 'pgclass'
      Total runtime for test 'pgclass': 0:00:00.21
      Performing test 'namespace'
      Total runtime for test 'namespace': 0:00:00.04
      Performing test 'distribution_policy'
      Total runtime for test 'distribution_policy': 0:00:00.01
      Performing test 'dependency'
      Total runtime for test 'dependency': 0:00:00.16
      Performing test 'part_integrity'
      Total runtime for test 'part_integrity': 0:00:00.05
      Performing test 'orphaned_toast_tables'
      Total runtime for test 'orphaned_toast_tables': 0:00:00.36
      Performing test 'aoseg_table'
      Total runtime for test 'aoseg_table': 0:00:00.00
      Performing test 'ao_lastrownums'
      Total runtime for test 'ao_lastrownums': 0:00:00.00
```
**RCA:**
Internally missing_extraneous test of gpcheckcat tries to fetch the oid from all the catalogue tables present in the database. For pg_description table, the oid is changed to objoid in 6x and 7x. As there is no column as oid currently in pg_description table, hence an error is reported.

```
sshirisha=# \d+ pg_description;
                               Table "pg_catalog.pg_description"
   Column    |  Type   | Collation | Nullable | Default | Storage  | Stats target | Description
-------------+---------+-----------+----------+---------+----------+--------------+-------------
 objoid      | oid     |           | not null |         | plain    |              |
 classoid    | oid     |           | not null |         | plain    |              |
 objsubid    | integer |           | not null |         | plain    |              |
 description | text    | C         | not null |         | extended |              |
Indexes:
    "pg_description_o_c_o_index" UNIQUE, btree (objoid, classoid, objsubid)
Access method: heap
```
**Fix:**
Catalogue tables which do not have a OID column are stored in a dictionary 'TableMainColumn' and are mapped an existing OID type attribute. For fixing this issue an entry is added to dictionary so that objoid is used instead of oid while running SQL queries.

Additionally, a behave test is also added to catch any further inconsistencies with pg_description tabl
## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Review by jnihal — COMMENTED** [review]
> This is a backport of https://github.com/greenplum-db/gpdb/pull/16760

I think we should rephrase this as it is not really a backport.

File: gpMgmt/test/behave/mgmt_utils/gpcheckcat.feature [hunk 1]
```diff
@@ -729,19 +729,24 @@ Feature: gpcheckcat tests
...
             | attrname   | tablename          |
             | relname    | pg_class           |
-
-    Scenario: gpcheckcat should discover missing attributes of pg_description catalogue table
-        Given there is a "heap" table "public.heap_table" in "miss_attr_db5" with data and description
-        When the user runs "gpcheckcat -v miss_attr_db5"
-        And gpcheckcat should return a return code of 0
-        Then gpcheckcat should not print "Missing" to stdout
-        And the user runs "psql miss_attr_db5 -c "SET allow_system_table_mods=true; DELETE FROM pg_description where objoid='heap_table'::regclass::oid;""
+    @shi
```
[the remark this answers is not in the document]
Comment (ShirishaRao) [reply gpMgmt/test/behave/mgmt_utils/gpcheckcat.feature]: removed!

**Review by shrakesh — APPROVED** [review]
LGTM
